keps/sig-multicluster/1645-multi-cluster-services-api/README.md

```diff
@@ -715,30 +716,24 @@ to describe a selector based policy that applies to a multi-cluster service.
 
 _Optional, but recommended._
 
-MCS aims to align with the existing [service DNS
+The full specification for Multicluster Service DNS is in this KEP's [specification.md](specification.md). MCS aims to align with the existing [service DNS
 spec](https://github.com/kubernetes/dns/blob/master/docs/specification.md). This
-section assumes familiarity with in-cluster Service DNS behavior.
-```
-<<[UNRESOLVED]>>
-Full DNS spec needed prior to Beta graduation following:
-https://github.com/kubernetes/dns/blob/master/docs/specification.md
-<<[UNRESOLVED]>>
-```
-When a `ServiceExport` is created, this will cause a domain name for the
+section provides an overview of the multicluster DNS specification and its rationale, and assumes familiarity with in-cluster Service DNS behavior.
+
+In short, when a `ServiceExport` is created, this will cause a domain name for the
 multi-cluster service to become accessible from within the clusterset. The
-domain name will be `<service>.<ns>.svc.clusterset.local`. 
+domain name will be `<service>.<ns>.svc.clusterset.local`. This domain name operates differently depending on whether the `ServiceExport` refers to a ClusterSetIP or Headless service:
 
-**ClusterSetIP services:** Requests to this domain name from within an importing
+  * **ClusterSetIP services:** Requests to this domain name from within an importing
 cluster will resolve to the clusterset IP, which points to endpoints for pods
 within the underlying `Service`(s) across the clusterset.
-
-**Headless services:** Within an importing cluster, the clusterset domain name
+  * **Headless services:** Within an importing cluster, the clusterset domain name
 will have multiple `A`/`AAAA` records, each containing the address of a ready
 endpoint of the headless service. `<service>.<ns>.svc.clusterset.local` will
 resolve to the set of all ready pod IPs for the service.
```

Comment:
nit: is it guaranteed to be ALL or just a strict subset of all?  Literal ALL could be quite large

Reply:
The [referenced e2e test](https://github.com/kubernetes/kubernetes/blob/master/test/e2e/network/dns_scale_records.go) only verifies clusterIP, it does not create headless services with a lage number of endpoints. With a large number of endpoints, the answer list might not fit in a single UDP response(I have not verified what the breaking point is), so it might be useful to have a test to see if 10k endpoints are handled. With multi-cluster, the endpoints count could be much larger than that, so I +1 "strict subset".

Maybe we can reword as -  "the DNS name will resolve to the entire set or the subset of ready pod IPs, depending on the implementation and endpoint count."?

Reply:
I like this suggested wording. I think implementations should decide how many to return and how to select them. Flexibility here is important and as mentioned above, it may not be feasible to return all endpoints. Clients balancing between a subset is still useful. I expect that most/all services that truly require clients to know ALL endpoints will have their own discovery mechanisms that take over after the initial connection.

Reply:
Fixed, used your wording in [5419c15](https://github.com/kubernetes/enhancements/pull/2577/commits/5419c150af279e1f1841cf95461dcad30ff1b4bb) 😇

Reply:
Plus added this to actual `specification.md` using wording in [693d15d](https://github.com/kubernetes/enhancements/pull/2577/commits/693d15d1af01ee7aeeb3b2ec90278e0cf2d026e9). FYI I said "subset" instead of "strict subset" (reiterating that "subset" _may_ be "all") and also require that pod IP records and AAAA records should be for the same subset of pod IPs.